lib/discourse_restricted_replies/guardian_extension.rb

```diff
@@ -0,0 +1,23 @@
+# frozen_string_literal: true
+
+module DiscourseRestrictedReplies
+  module GuardianExtension
+    extend ActiveSupport::Concern
+
+    def can_create_post_on_topic?(topic)
+      return super if !SiteSetting.restricted_replies_enabled
```

Comment:
Are `enabled` checks required in class extensions?

Reply:
When adding methods to existing classes, we should check for the enabled setting, yes. It’s not done automatically when extending “manually” a class, so we should probably keep the existing checks when they exist. (Maybe at some point we could do something to automatically add the checks)